catalog-rest-service/src/main/java/org/openmetadata/catalog/security/SecurityUtil.java

```diff
@@ -56,17 +57,28 @@ public static void checkAdminRoleOrPermissions(
     }
   }
 
+  /**
+   * Most REST API requests should yield in a single metadata operation. There are cases where the JSON patch request
+   * may yield multiple metadata operations. This helper function checks if user has permission to perform the given set
+   * of metadata operations.
+   */
   public static void checkAdminRoleOrPermissions(
       Authorizer authorizer,
       SecurityContext securityContext,
       EntityReference entityReference,
-      MetadataOperation metadataOperation) {
+      List<MetadataOperation> metadataOperations) {
+
     Principal principal = securityContext.getUserPrincipal();
     AuthenticationContext authenticationCtx = SecurityUtil.getAuthenticationContext(principal);
-    if (!authorizer.isAdmin(authenticationCtx)
-        && !authorizer.isBot(authenticationCtx)
-        && !authorizer.hasPermissions(authenticationCtx, entityReference, metadataOperation)) {
-      throw new AuthorizationException("Principal: " + principal + " does not have permissions");
+
+    if (authorizer.isAdmin(authenticationCtx)) return;
```

Comment:
can you merge these twos if statements into a single one

Reply:
Done